sunpy/util/scraper.py

```diff
@@ -343,22 +365,74 @@ def _extract_files_meta(self, timerange, extractor=None):
             Time interval where to find the directories for a given pattern.
         extractor: `str`
             Pattern to extract metadata by parsing the URL.
+        matcher: `dict`
+            Dictionary to check if extracted metadata is valid.
```

Comment:
Is there any test with the `matcher` I don't really get what it does.

Reply:
Changed on test which uses matcher. `matcher` is a dict which contains a list of possible values that a url can have in that place. Say if url pattern `domain/{goes_no}_goes_file.fits` can return files for `goes_no` 1,2 and 3. If we do 
`matcher = {'goes_no':[1,2]}` then goes_no 3 files won't be returned.